common/src/sm.c

```diff
@@ -0,0 +1,584 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stdlib.h>
+#include <stdint.h> // IWYU pragma: keep
+#include <inttypes.h> // IWYU pragma: keep
+#include <stdbool.h>
+#include <string.h>
+
+#include "macro_utils/macro_utils.h"
+
+#include "c_logging/xlogging.h"
+#include "c_pal/gballoc_hl.h" // IWYU pragma: keep
+#include "c_pal/gballoc_hl_redirect.h"
+#include "c_pal/interlocked.h"
+#include "c_pal/threadapi.h"
```

Comment:
>threadapi [](http://example.com/codeflow?start=16&length=9)

looks like threadapi should also go in the pal_ll

Reply:
which makes sense because it has no other dependencies really

Reply:
moved to pal_ll now